CustomPlugin/ServiceClientFactory.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using CustomPlugin.Environment;
+using Microsoft.AspNetCore.Http;
+using Service;
+
+namespace CustomPlugin
+{
+    public class ServiceClientFactory
+    {
+        private readonly MiaEnvConfiguration _miaEnvConfiguration;
+        private static  HttpRequestHeaders _miaHeaders = ServiceProxy.GetDefaultHeaders();
+        private static HttpRequestHeaders MiaHeaders => _miaHeaders;
+
+        public static void SetMiaHeaders(MiaHeadersPropagator miaHeadersPropagator)
+        {
+            _miaHeaders = ServiceProxy.GetDefaultHeaders();
+
+            foreach (var (key, value) in miaHeadersPropagator.Headers)
+            {
+                MiaHeaders.Remove(key);
```

Comment:
Isn't there a `Set` method that overwrite previous value instead of removing and adding keys?

Reply:
Unfortunately there isn't, I searched for it but I didn't find any way to to this in a different way. We have to make sure that the header isn't already there because otherwise we would concatenate the added value to the current value.